packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -4185,4 +4188,19 @@ export class SnapController extends BaseController<
       runtime.state = undefined;
     }
   }
+
+  /**
+   * Track usage of a Snap by updating last interaction with it.
+   * Note: Interaction tracking is done only for specific handler types.
+   *
+   * @param snapId - ID of a Snap.
+   * @param handlerType - Type of Snap handler.
+   */
+  updateLastInteraction(snapId: SnapId, handlerType: HandlerType): void {
```

Comment:
We already have some code that tracks this, we just don't persist it in state, it is only stored in memory: https://github.com/MetaMask/snaps/blob/4b6dadbc8d691e614f2d83c5fe0afe23ee665039/packages/snaps-controllers/src/snaps/SnapController.ts#L3807-L3822

Side-note: I wonder how many extra state updates this will cause? May be somewhat excessive and cause load on the clients 🤔